cartridge/pool.lua

```diff
@@ -203,29 +195,57 @@ local function map_call(fn_name, args, opts)
         uri_map[uri] = true
     end
 
-    local maps = {}
-    local fibers = {}
-    for _, uri in pairs(opts.uri_list) do
-        local fiber = fiber.new(
-            _gather_netbox_call,
-            fiber.self().storage, maps, uri, fn_name, args,
-            {timeout = opts.timeout}
-        )
-        fiber:name('netbox_map_call')
-        fiber:set_joinable(true)
-        fibers[uri] = fiber
+    local maps = {nil, nil}
+    local fibers = table.new(0, #opts.uri_list)
+    local futures = table.new(0, #opts.uri_list)
+
+    local timeout = opts.timeout or 10
+    local deadline = fiber.clock() + timeout
+
+    for _, uri in ipairs(opts.uri_list) do
+        local conn, err = connect(uri, {wait_connected = false})
+        if conn == nil then
+            _pack_values(maps, uri, nil, err)
+        elseif conn:is_connected() then
+            local future, err = errors.netbox_call(
+                conn, fn_name, args, {is_async = true}
+            )
```

Comment:
Can it ever raise an error?

Reply:
For ex. connection became broken (but it's unlikely).
Also, I think it's better to wrap all `netbox` methods.